### PR TITLE
Feature/delete users

### PR DESCRIPTION
**Pull request by stanislaukhamiankou**

### CODE REVIEW

File: src/components/_molecules/ButtonDeleteUser/ButtonDeleteUserType.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export type ButtonDeleteUserType = {
+  onConfirm:
+    | ((e?: React.MouseEvent<HTMLElement, MouseEvent> | undefined) => void)
+    | undefined
```
Comment (Chubarava) [inline src/components/_molecules/ButtonDeleteUser/ButtonDeleteUserType.d.ts:4]: why do you use undefined?

File: src/components/_organisms/Table/Table.tsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 
 import { TableType } from './Table.d'
 
-export const List = ({ className, data }: TableType): JSX.Element => (
-  <div className={cn('user-list', className)}>
-    <Table dataSource={data} size="middle" columns={columns} bordered />
-  </div>
-)
+export const List = ({ className, data }: TableType) => {
+  const columns = Columns()
```
Comment (AlehSabadashGmail) [inline src/components/_organisms/Table/Table.tsx:8]: it is not reusable if you put Columns here
Table is just component and you need to reuse it in different places with different columns

File: src/components/_molecules/ButtonDeleteUser/ButtonDeleteUser.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { DeleteOutlined } from '@ant-design/icons'
+import { Button, Popconfirm } from 'antd'
+import { ButtonDeleteUserType } from './ButtonDeleteUserType'
+
+export const ButtonDeleteUser = ({ onConfirm }: ButtonDeleteUserType) => {
+  return (
+    <Popconfirm title="Are you sure?" onConfirm={onConfirm}>
+      <Button icon={<DeleteOutlined />} />
```
Comment (AlehSabadashGmail) [inline src/components/_molecules/ButtonDeleteUser/ButtonDeleteUser.tsx:8]: I think we will be reused it in different places, so change the name of component

File: src/constants/Columns/Column.tsx [hunk 4]
```diff
@@ -1,83 +1,108 @@
...
-        </>
-      )
+export const Columns = () => {
+  const dispatch = useAppDispatch()
+
+  const deleteUser = (id: string) => {
+    dispatch(deleteUsers(id))
+  }
+
+  const columns: ColumnsType<IUser> = [
```
Comment (AlehSabadashGmail) [inline src/constants/Columns/Column.tsx:18]: thing about optimisation, because columns will be created each render

**Pull request closed without merging**